pkg/controllers/sync/controller.go

```diff
@@ -687,95 +629,28 @@ func (s *SyncController) setFederatedStatus(ctx context.Context, fedResource Fed
 		return worker.StatusError
 	}
 
-	// UpdateStatus does not read the annotations, only the status field.
-	// Update reads the annotations, but it will bump the generation if status is changed.
-	// Therefore, we have to separate status update and annotation update into two separate calls.
-
-	err = wait.PollImmediate(1*time.Second, 5*time.Second, func() (bool, error) {
-		changed := false
-		err := sourcefeedback.PopulateSyncingAnnotation(obj, collectedStatus.StatusMap, &changed)
-		if err != nil {
-			return false, err
-		}
-
-		if !changed {
-			return true, nil
-		}
-
-		err = s.hostClusterClient.Update(context.TODO(), obj)
-		if err == nil {
-			return true, nil
-		}
-
-		if apierrors.IsConflict(err) {
-			err := s.hostClusterClient.Get(context.TODO(), obj, obj.GetNamespace(), obj.GetName())
-			if err != nil {
-				return false, errors.Wrapf(err, "failed to retrieve resource")
-			}
-			return false, nil
-		}
-		return false, errors.Wrapf(err, "failed to update resource")
-	})
-	if err != nil {
-		keyedLogger.Error(err, "Failed to update syncing annotation")
-		return worker.StatusError
-	}
-
 	return worker.StatusAllOK
 }
 
 func (s *SyncController) ensureDeletion(ctx context.Context, fedResource FederatedResource) worker.Result {
```

Comment:
```suggestion
func (s *SyncController) handleTerminatingFederatedResource(ctx context.Context, fedResource FederatedResource) worker.Result {
```

Reply:
Let's rename the methods used for handling deletion to be more aligned with what they do:

`ensureDeletion` -> `handleTerminatingFederatedResource`
`deleteFromClusters` -> `ensureRemovalFromClusters`
`handleDeletionFromClusters` -> `removeFromAllClusters`
`deleteFromCluster` -> `removeFromCluster`
`ensureRemovedOrUnmanaged` -> `checkObjectRemovedFromAllClusters`

Reply:
Renamed all but `handleDeletionFromClusters` -> `removeFromAllClusters`. `handleDeletionFromClusters` invokes a custom `deletionFunc` and should be named differently from `removeFromCluster`.